cmd/encryption-v1.go

```diff
@@ -643,7 +623,7 @@ func DecryptBlocksRequest(client io.Writer, r *http.Request, bucket, object stri
 	var seqNumber uint32
 	var encStartOffset, encLength int64
 
-	if len(objInfo.Parts) == 0 || !objInfo.IsEncryptedMultipart() {
+	if len(objInfo.Parts) == 0 || !crypto.IsMultiPart(objInfo.UserDefined) {
```

Comment:
I also discourage having crypto being aware about backend data format

Reply:
@vadmeste That was done on purpose. That's one of the improvements we get from extracting this out into `crypto`
Otherwise we cannot simplify and simplification *is* needed.